app/controllers/core_data_connector/projects_controller.rb

```diff
@@ -66,38 +62,8 @@ def import_data
       project = Project.find(params[:id])
       authorize project, :import_data?
 
-      begin
-        # Create the target directory
-        destination = "#{Rails.root}/tmp/#{SecureRandom.urlsafe_base64}"
-        FileUtils.mkdir_p(destination) unless File.exist? destination
-
-        # Extract the zip file to the directory
-        Zip::File.open(params[:file].tempfile.path) do |zipfile|
-          zipfile.each do |entry|
-            # Ignore directories
-            next unless entry.file?
-
-            # Ignore MacOS archive
-            next if entry.name =~ IGNORE_PATTERN
-
-            # Extract the file
-            zipfile.extract(entry, File.join(destination, entry.name))
-          end
-        end
-
-        # Create a new importer with the temp directory and run it
-        ActiveRecord::Base.transaction do
-          importer = Import::Importer.new(destination)
-          importer.run
-        end
-
-        # Remove the temporary directory
-        FileUtils.rm_rf(destination)
-      rescue ActiveRecord::RecordInvalid => exception
-        errors = [exception]
-      rescue StandardError => exception
-        errors = [exception]
-      end
+      zip_importer = CoreDataConnector::Import::ZipHelper.new
```

Comment:
Since this class is also in the `CoreDataConnector` module, I think you could instantiate the object with just `Import::ZipHelper.new`.